src/DataProvider/SubresourceDataProviderInterface.php

```diff
@@ -32,7 +32,7 @@
      *
      * @throws ResourceClassNotSupportedException
      *
-     * @return object|null
+     * @return object|array|null
```

Comment:
You also have to change the PHPDoc for:
* `ApiPlatform\Core\GraphQl\Resolver\Factory\CollectionResolverFactory::getSubresource()`
* `ApiPlatform\Core\DataProvider\OperationDataProviderTrait::getSubresourceData()`